repository/src/main/java/com/epam/esm/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.UserDao;
+import com.epam.esm.entity.User;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private static final String PAGE = "page";
+    private static final String PER_PAGE = "per_page";
+    private static final String SELECT_ALL_USERS = "FROM User";
+    @PersistenceContext
+    private EntityManager entityManager;
+
+
+    @Override
+    public Optional<User> findById(Long id) {
```

Comment:
Why Optional?

Reply:
because a User with such a id may not exist. And then the error is thrown